### PR TITLE
feat: Mark posts as read after a delay of 2 seconds [BD-38] [TNL-8658] [BB-4734]

### PR DESCRIPTION
**Pull request by xitij2000**
This commit adds support for automatically marking a post as read after it's been open for 2 seconds. The delay value is customisable.

### CODE REVIEW

**Comment by tinumide** [comment]
:+1:

- [x] I tested that and checked that posts are marked as read after 2 seconds of delay
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] ~Includes documentation~ NA

**Review by tinumide — APPROVED** [review]
:+1:

- [x] I tested that and checked that posts are marked as read after 2 seconds of delay
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] ~Includes documentation~ NA

File: src/discussions/comments/CommentsView.jsx [hunk 1]
```diff
@@ -3,23 +3,35 @@ import React, { useEffect } from 'react';
...
   useEffect(() => {
     dispatch(fetchThreadComments(postId));
+    const markReadTimer = setTimeout(() => {
+      if (thread && !thread.read) {
+        dispatch(markThreadAsRead(postId));
+      }
+    }, getConfig().POST_MARK_AS_READ_DELAY);
+    return () => {
+      clearTimeout(markReadTimer);
+    };
```
Comment (mehaknasir) [inline src/discussions/comments/CommentsView.jsx:34]: if we only have to dispatch one call under set timeout which is conditional, then can we shift the if statement above like this?

`if (thread && !thread.read) {
        const markReadTimer = setTimeout(() => {
            dispatch(markThreadAsRead(postId));
        }, getConfig().POST_MARK_AS_READ_DELAY);
        return () => {
          clearTimeout(markReadTimer);
        };
}`

File: .env [hunk 2]
```diff
@@ -14,6 +14,7 @@ LOGO_WHITE_URL=''
 FAVICON_URL=''
 MARKETING_SITE_BASE_URL=''
 ORDER_HISTORY_URL=''
+POST_MARK_AS_READ_DELAY=null
```
Comment (mehaknasir) [inline .env:17]: why passing null here?

Comment (xitij2000) [reply .env:17]: A blank string didn't seem to match the type here, which is a number.

File: src/discussions/comments/CommentsView.jsx (hunk 1 above)
Comment (xitij2000) [reply src/discussions/comments/CommentsView.jsx:34]: I think the code got a bit jumbled above, but if I understand the intention right you'd like to make the setTimeout and return conditional? 

The  issue there is that either react or the linter doesn't like this and recommends having a return for both paths, which mean we'd need a additional bogus return statement which makes the code less clean.

Comment (mehaknasir) [reply src/discussions/comments/CommentsView.jsx:34]: yes, my intention here is that if we don't need to dispatch the call then put all the code under condition.  I am not sure about your statement here. but I am good with it if you feel so.

File: .env (hunk 2 above)
Comment (mehaknasir) [reply .env:17]: why not define the default value here?

Comment (xitij2000) [reply .env:17]: Will do.

**Review by mehaknasir — APPROVED** [review]

**Review by awais-ansari — APPROVED** [review]

**Comment by edx-webhook** [comment]
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-app-discussions&entry.867055334=https://github.com/edx/frontend-app-discussions/pull/11&entry.1484655318=https://github.com/xitij2000&entry.752974735=2021-08-26+11:21&entry.1917517419=2021-09-07+15:38&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**